src/Core/Services/ExecutionHelper.cs

```diff
@@ -0,0 +1,596 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
```

Comment:
Please update the PR description to reflect the kind of changes in this PR, and how it was tested.